src/core/server/types.ts

```diff
@@ -18,7 +18,7 @@
  */
 
 /** This module is intended for consumption by public to avoid import issues with server-side code */
-export { PluginOpaqueId } from './plugins/types';
+export { PluginOpaqueId, SharedGlobalConfig } from './plugins/types';
```

Comment:
Good catch, we forgot to expose that in https://github.com/elastic/kibana/pull/51478.

However `core/server/types` should only expose types that may be used by the client-side, which is not the case of `SharedGlobalConfig`.

If you don't mind, could you please:
- Export it here instead
https://github.com/elastic/kibana/blob/bb7c5f00e51ee066a9edfc54b65aef3a0770921c/src/core/server/index.ts#L134-L143
- Add a `/** @public */` on the type
https://github.com/elastic/kibana/blob/bd3857a79befd9baa88e0ca76f6b1594f8716b88/src/core/server/plugins/types.ts#L209-L213
- And regenerate core doc as a new type is exported

`node scripts/check_core_api_changes.js --accept --docs` - then commit the result (be careful to preserve windows line return as doc is generated with CRLF)

If you prefer, I can do it and push on your PR branch

Reply:
If you don't mind pushing against my PR branch, that'd be really helpful. I tried doing those things and now I'm getting some new warnings in server.api.md, and I'd just prefer someone who knows more about what they're doing take care of it 😬

Reply:
Ack, will do today